ceph/ceph.py

```diff
@@ -1913,6 +1916,606 @@ def enable_ceph_mgr_restful(self):
+        """
+        Read cephadm generated public key
+        Arg:
+            Installer node
+        Returns:
+            Public Key string
+        """
+        ceph_pub_key, _ = self.exec_command(cmd="sudo cat /etc/ceph/ceph.pub")
+        return ceph_pub_key.read().decode().strip()
+
+    def distribute_cephadm_gen_pub_key(self, ceph_cluster, nodes=None):
+        """
+        Distribute cephadm generated public key to all nodes in the list.
+        Args:
+            ceph_cluster: ceph cluster object
+            nodes: node list to add ceph public key(default: None)
+        """
+        ceph_pub_key = self.read_cephadm_gen_pub_key()
+
+        if nodes is None:
+            nodes = ceph_cluster.get_nodes()
+
+        nodes = nodes if isinstance(nodes, list) else [nodes]
+
+        for each_node in nodes:
+            each_node.exec_command(cmd="sudo install -d -m 0700 /root/.ssh")
+            keys_file = each_node.write_file(
+                sudo=True,
+                file_name='/root/.ssh/authorized_keys', file_mode='a'
+            )
+            keys_file.write(ceph_pub_key)
+            keys_file.flush()
+            each_node.exec_command(
+                cmd="sudo chmod 0600 /root/.ssh/authorized_keys"
+            )
+
+
+class CephAdmin(object):
+    """
+    Ceph Administrator
+    """
+    ROLES = ["mon", "mgr", "osd", "mds", "rgw", "iscsi", "client"]
+    TIMEOUT = 300
+
+    def __init__(self, name, ceph_cluster, ceph_installer, **kwargs):
+        self.cluster = ceph_cluster
+        self.cluster_name = name
+        self.installer = ceph_installer
+        self.config = kwargs
+        self.first_mon = self.installer.node
+        self.first_mgr = self.installer.node
+        self.image = None
+
+    def image(self):
+        return self.image
+
+    def repo(self):
+        return self.config.get('base_url')
+
+    def check_exist(self, daemon, ids, timeout=None, interval=5):
+        """
+        validate daemon existence
+        Args:
+            daemon: daemon name
+            ids: daemon ids
+            timeout: timeout in seconds
+            interval: interval in seconds
+        Returns:
+            boolean
+        """
+        timeout = self.TIMEOUT if not timeout else timeout
+        checks = timeout / interval
+
+        while checks:
+            checks -= 1
+
+            out, _ = self.shell(
+                remote=self.installer,
+                args=["ceph", "orch", "ps", "-f", "json-pretty"],
+            )
+
+            out = json.loads(out)
+            daemons = dict((i['daemon_id'], i['status_desc'])
+                           for i in out if i.get('daemon_type') == daemon)
+
+            count = 0
+            for _id in ids:
+                if _id in daemons:
+                    status = daemons[_id]
+                    if status == 'running':
+                        count += 1
+
+            logger.info("{up_count}/{num} {daemon}s up....".format(num=len(ids),
```

Comment:
Avoid using formatter with logging instead use `%`... read about it [doc](https://stackoverflow.com/questions/34619790/pylint-message-logging-format-interpolation)

Reply:
though the advantage here is interpreting the strings when pylint is being used. but we have are not validating anything against pylint. if we have to formalise it we need to format the entire code base and emit these kinda messages. there are places we have `%` and there are places we have used `format` and there are places we have `f'{ }`

Reply:
Agreed. I am also not in favor of refactoring... at some point in the future we would need to move to a standard based on consensus. @sunilkumarn417 you can chose to ignore this comment.